### DISCUSSION TITLE
Establishing a Document Hierarchy in eForms for Parent-Child Relationships

### DISCUSSION THREAD

**Discussion by sebastian-hermix** [OP-TED/eForms-SDK#681] [Q&A] [answered]
In the context of eForms, there is a need to establish a document hierarchy like this one(https://ted.europa.eu/udl?uri=TED:NOTICE:574383-2023:TEXT:EN:HTML&tabId=4) that reflects parent-child relationships. However, it appears that the existing attributes, such as the `cbc:ContractFolderID` tag, primarily function by aggregating notices into a folder without providing explicit parent-child attributes. This limitation makes it challenging to reconstruct a detailed document tree from multiple documents.

The current setup, utilizing the `cbc:ContractFolderID` tag, effectively groups notices into a common folder. However, it does not offer the nuanced parent-child relationships required for creating a more complex document structure.

My questions are: 
1.  Is there currently a tag or method available to establish a document hierarchy, similar to using a `PARENT_ID` tag? 

2.    If such a feature does not exist at present, is there any plan or indication that it might be introduced in the near future?

**Comment by bertrand-lorentz** [comment, marked as answer]
Please note that, in the link you provide, all notices shown are eForms notices.

I think the TED website creates this document hierarchy by using the following information (I'd need to check with my colleagues to confirm):

- Get all notices in a procedure based on the "BT-04-notice" field ( the `cbc:ContractFolderID` element), as you indicated
- Change notice reference the notice being changed via the field "BT-758-notice" (`efbc:ChangedNoticeIdentifier`), so this can be used for a relationship.
- Contract Award notices are distinguished based on the notice/form type or notice subtype.

What other type of relationship would you like to represent ? And what information is missing in order to do it ?

**Reply by sebastian-hermix** [reply]
In the absence of the cbc:ContractFolderID field within the Prior Information Notice (PIN), it becomes uncertain how to create clear connections with other procurement documents.

**Reply by bertrand-lorentz** [reply]
Ah, yes, I had forgotten about this relation.

PIN notices are not part of a procedure, as there is no real procedure when a PIN is published. Later on, there might not be any, or there might be several procedures.
In the competition notice, the author can indicate the identifier of the PIN that is related with the notice, in field "BT-125(i)-Lot". This field is optional, as there might not have been any related PIN.
